tesseract_environment/src/utils.cpp

```diff
@@ -306,13 +273,22 @@ bool checkTrajectory(std::vector<tesseract_collision::ContactResultMap>& contact
 
         for (long iSubStep = start_idx; iSubStep < end_idx; ++iSubStep)
         {
+          tesseract_collision::ContactTrajectorySubstepResults substep_contacts(static_cast<int>(iSubStep),
```

Comment:
I would not define the size here. This is one reason for the performance hit because it will allocate memory when it is not needed. Do this in the if statement below.

Reply:
This is not actually allocating a vector, it is just setting the substep number which is an int.
```
ContactTrajectorySubstepResults::ContactTrajectorySubstepResults(const int& substep_number, Eigen::VectorXd state)
  : substep(substep_number), state0(std::move(state)), state1(std::move(state))
{
}
```

Reply:
but it does result in dynamic memory allocation because using Eigen::VectorXd which is not a fixed size object which can be avoided if it is only constructed if debug is enabled.